### DISCUSSION TITLE
Consider making semicolon delineation stricter

### DISCUSSION THREAD

**Discussion by Fishrock123** [kdl-org/kdl#43] [Ideas]
(This is coming from the following twitter discussion: https://twitter.com/Fishrock123/status/1340133885988937728)

I think that being hand-wavy on newlines vs semicolons risks people not choosing one or the other, leading to JavaScript-esque style conflicts.

In my opinion, new languages should be prescriptive about their syntax and not overlap their features when possible, so as to decrease confusion.

----

As such, I propose that the spec be amended to only allow delineation / line continuation / "next node" via a semicolon only if the semicolon is followed by a whitespace (**excluding line breaks**) and then another node.

That is, keep this valid:
```kdl
node; node
```

but make this invalid:
```kdl
node;
node
```

**Comment by Fishrock123** [comment]
(Though personally I don't see much use in semicolon delineation here to begin with, this is a bit of a compromise.)

**Comment by larsgw** [comment]
I don't really like semicolons either and I feel the use case for data exchange is perhaps a bit limited: why would the serializer bother with encoding the special case of a list of empty nodes? I can think of two reasons (minified files and newline-delimited KDL) but both require semicolons for non-empty nodes as well. Maybe I am missing something though.
